[Docs] Added detail to Terrain Macro Material. (#2155)
o3de/o3de.org: content/docs/user-guide/components/reference/terrain/terrain-macro-material.md

weight: 100
---

The **Terrain Macro Material** component applies a terrain base color texture to all terrain regions that fall within its volume.

> ## Usage

Reviewer: Can you move the Usage section to be below the the Properties section? This is consistent with other component reference pages like terrain world, terrain world renderer, and others. (There are some that still have Usage at the top, but we are fixing those along the way.)

Reviewer: Between Properties and EBus definitions!

Author: Done!

> The **Terrain Macro Material** component provides low-fidelity color information to the terrain system within a volume. The color data is used as the only source of color for distant terrain beyond the **Detail material render distance** defined on the [**Terrain World Renderer**](/docs/user-guide/components/reference/terrain/world-renderer) component. Within the **Detail material render distance** the color data is blended with the [terrain detail materials](/docs/user-guide/components/reference/terrain/terrain-detail-material) to provide color variation as the detail textures repeat on the surface.

Reviewer: ```suggestion
The **Terrain Macro Material** component provides low-fidelity color information to the terrain system within a volume. The color data is used as the only source of color for distant terrain that's positioned beyond the **Detail material render distance** defined on the [**Terrain World Renderer**](/docs/user-guide/components/reference/terrain/world-renderer) component. Within the **Detail material render distance** the color data is blended with the [terrain detail materials](/docs/user-guide/components/reference/terrain/terrain-detail-material) to provide color variation as the detail textures repeat on the surface.
```

Author: Hm. The phrase "that's positioned" makes it sound too discrete to me, like the terrain either exists near or far, instead of stretching from near to far. How about this?

```suggestion
The **Terrain Macro Material** component provides low-fidelity color information to the terrain system within a volume. The color data is used as the only source of color for any portion of the terrain that exists beyond the **Detail material render distance** defined on the [**Terrain World Renderer**](/docs/user-guide/components/reference/terrain/world-renderer) component. Within the **Detail material render distance** the color data is blended with the [terrain detail materials](/docs/user-guide/components/reference/terrain/terrain-detail-material) to provide color variation as the detail textures repeat on the surface.
```

The component also provides optional macro normal information to the terrain system so that distant terrain can have higher quality normals on the lower-poly LODs. The macro normals are difficult to author correctly, so they aren't recommended for typical use. The macro normals must be in world space and generated at the same terrain scale in all dimensions as the terrain in the O3DE level. If the terrain in O3DE is resized unevenly in any direction, the normals will no longer line up with the terrain geometry and the terrain surface lighting will be incorrect.

> You configure the dimensions of the volume by adjusting the [Axis Aligned Box Shape](/docs/user-guide/components/reference/shape/axis-aligned-box-shape) component on the entity. You assign a color texture and an optional normal texture by dragging texture assets to the **Color Texture** or **Normal Texture** fields, or by clicking {{< icon "file-folder.svg" >}}. Once the textures are assigned, you can adjust the direction and magnitude of the normals by using the **Normal Flip X** and **Normal Flip Y** toggles and the **Normal Factor** slider.

Reviewer: I know this text is unchanged from an earlier version, but I have a few suggestions. 
```suggestion
You can configure the dimensions of the volume by adjusting the [Axis Aligned Box Shape](/docs/user-guide/components/reference/shape/axis-aligned-box-shape) component on the entity. You can assign a color texture and an optional normal texture by dragging texture assets to the **Color Texture** or **Normal Texture** fields, or by clicking {{< icon "file-folder.svg" >}}. When the textures are assigned, you can adjust the direction and magnitude of the normals by using the **Normal Flip X** and **Normal Flip Y** toggles, and the **Normal Factor** slider.
```

Author: Is that final comma correct? With the way it's written, it's only a list of two items - "by using the toggles, and the slider" - which I didn't think warrants a comma. I thought the comma only belongs if it were written as "by using the X toggle, the Y toggle, and the Factor slider".
